[RUM] add documentation for loading time and loading type (#7680)
DataDog/documentation: content/en/real_user_monitoring/data_collected/view.md

For page views, loading performance metrics are collected from both the [Paint Timing API][5] and the [Navigation Timing API][6].

## Single Page Applications

> For Single Page Applications (SPAs), the RUM SDK differentiates between `initial_load` and `route_change` navigations with the `loading_type` attribute. If a click on your web page leads to a new page without a full refresh of the page, the RUM SDK will start a new View event with `loading_type:route_change`. RUM is able to track page changes using the [History API][7].

Reviewer: ```suggestion
For Single Page Applications (SPAs), the RUM SDK differentiates between `initial_load` and `route_change` navigations with the `loading_type` attribute. If a click on your web page leads to a new page without a full refresh of the page, the RUM SDK starts a new view event with `loading_type:route_change`. RUM tracks page changes using the [History API][7].
```

> Datadog provides a unique performance metric, Loading Time, which calculates the time needed for a page to load. This metric works both for `initial_load` and `route_change` navigations.

Reviewer: ```suggestion
Datadog provides a unique performance metric, `loading_time`, which calculates the time needed for a page to load. This metric works both for `initial_load` and `route_change` navigations.
```

> ### How is Loading Time calculated?

Reviewer: ```suggestion
### How is loading time calculated?
```

> To account for modern web applications, Loading Time watches for network requests and long tasks.

Reviewer: ```suggestion
To account for modern web applications, loading time watches for network requests and long tasks.
```

> * **Initial Load**: Loading Time is equal to the difference between `navigationStart` and:

Reviewer: ```suggestion
* **Initial Load**: loading time is equal to the difference between `navigationStart` and the load event end.
```

Author: It's either the load event end or no activity on the page for 100ms

>   * the load event end\

Reviewer: ```suggestion
```

*or, whichever comes last,*

>   *  the first time the page is idle for more than 100ms (idle being defined as: no network requests, and no long tasks are currently occurring)

Reviewer: ```suggestion
  *  the first time the page is idle for more than 100ms (idle being defined as no network requests, and no long tasks are currently occurring)
```

> * **SPA Route Change**: Loading Time is equal to the difference between the user click and the first time the page is idle for more than 100ms (idle being defined as: no network requests, and no long tasks are currently occurring)

Reviewer: ```suggestion
* **SPA Route Change**: Loading Time is equal to the difference between the user click and the first time the page is idle for more than 100ms (idle being defined as no network requests, and no long tasks are currently occurring)
```

## Metrics collected

{{< img src="real_user_monitoring/data_collected/view/timing_overview.png" alt="Timing overview"  >}}

| Attribute                              | Type        | Decription                                                                                                                                                                                                                 |
|----------------------------------------|-------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `duration`                             | number (ns) | Time spent on the current view.                                                                                                                                                                                                  |

> | `view.loading_time`                             | number (ns) | Time until the page is ready and no network or long task activity is currently occurring. [Learn more][8]|

Reviewer: ```suggestion
| `view.loading_time`                             | number (ns) | Time until the page is ready and no network or long task activity is currently occurring. [More info from the data collected documentation][8]|
```

> | `view.measures.first_contentful_paint` | number (ns) | Time when the browser first rendered any text, image (including background images), non-white canvas, or SVG.[Learn more][9]                                                                                                |

Reviewer: ```suggestion
| `view.measures.first_contentful_paint` | number (ns) | Time when the browser first rendered any text, image (including background images), non-white canvas, or SVG. For more information about browser rendering, see the [w3 definition][9].                                                                                            |
```
